components/dashboard/src/utils.ts

```diff
@@ -85,6 +85,31 @@ export function getURLHash() {
     return window.location.hash.replace(/^[#/]+/, "");
 }
 
+export type DeepPartial<T> = {
+    [P in keyof T]?: DeepPartial<T[P]> | T[P];
+};
+
+function isObject(item: any): item is Record<string, any> {
+    return item && typeof item === "object" && !Array.isArray(item);
+}
+
+export function deepMerge<T>(target: T, source: DeepPartial<T>): T {
```

Comment:
Would rather we pull in something like `lodash.merge` for this and use it directly. Lots of weird edge cases w/ deep merging objects that would be nice to avoid (wrote [this npm module](https://www.npmjs.com/package/deap) years ago and learned about them).

Reply:
Good call! Thanks for calling this out, I almost forgot I added this 🤦

Reply:
Ended up using the `deepmerge` package, which we already used somewhere else in the codebase.